### PR TITLE
feat: #tno-2054 - new page for MMI Products

### PR DESCRIPTION
**Pull request by kylgarmor**
- added new MMI Products page
- users can now toggle their subscription on products
- check box not updating though ???
- sneaky update to the MMI logo files
- reshuffled a lot of files in TNO-Core to make locations more sane

### CODE REVIEW

File: libs/net/dal/Services/ProductService.cs [hunk 1]
```diff
@@ -163,16 +176,21 @@ public override Product Update(Product entity)
     /// </summary>
     /// <param name="userId"></param>
     /// <returns></returns>
-    public async Task<int> Unsubscribe(int userId, int productId)
+    public async Task<Product> Unsubscribe(int userId, int productId)
```
Comment (Fosol) [inline libs/net/dal/Services/ProductService.cs:179]: async issue

File: libs/net/dal/Services/ProductService.cs [hunk 2]
```diff
@@ -163,16 +176,21 @@ public override Product Update(Product entity)
...
-            this.Context.Entry(s).State = EntityState.Deleted;
+            s.IsSubscribed = false;
+            if (!saveChanges) saveChanges = true;
         });
-        return saveChanges ? await Context.SaveChangesAsync() : await Task.FromResult(0);
+        this.Context.UpdateRange(userProducts);
+        this.CommitTransaction();
+
+        // return saveChanges ? await Context.SaveChangesAsync() : await Task.FromResult(0);
+        return FindById(productId);
```
Comment (Fosol) [inline libs/net/dal/Services/ProductService.cs:193]: Handle null values

File: libs/net/dal/Services/ProductService.cs [hunk 3]
```diff
@@ -181,26 +199,27 @@ public async Task<int> Unsubscribe(int userId, int productId)
     /// <param name="userId"></param>
     /// <param name="productId"></param>
     /// <returns></returns>
-    public async Task<int> Subscribe(int userId, int productId)
+    public async Task<Product> Subscribe(int userId, int productId)
```
Comment (Fosol) [inline libs/net/dal/Services/ProductService.cs:202]: Async issue

File: libs/net/dal/Services/ProductService.cs [hunk 4]
```diff
@@ -181,26 +199,27 @@ public async Task<int> Unsubscribe(int userId, int productId)
...
+            this.Context.UpdateRange(userProducts);
+            this.CommitTransaction();
         } else {
             this.Context.UserProducts.Add(new UserProduct(userId, productId, true));
-            saveChanges = true;
+            this.CommitTransaction();
         }
-        return saveChanges ? await Context.SaveChangesAsync() : await Task.FromResult(0);
+
+        return FindById(productId);
```
Comment (Fosol) [inline libs/net/dal/Services/ProductService.cs:222]: Handle null values

File: app/subscriber/src/features/my-products/MyProducts.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+export const MyProducts: React.FC = () => {
+  const [{ getProducts, toggleSubscription }] = useProducts();
+  const { toggle, isShowing } = useModal();
+  const [{ requests }] = useAppStore();
+
+  const [products, setProducts] = React.useState<IProductSubscriberModel[]>([]);
+  const [product, setProduct] = React.useState<IProductSubscriberModel>();
+
+  React.useEffect(() => {
+    getProducts().then((data) => {
```
[the remark this answers is not in the document]
Comment (kylgarmor) [reply app/subscriber/src/features/my-products/MyProducts.tsx:20]: I would counter with "Is this a page that many users will visiting multiple time during a single session?".

[the remark this answers is not in the document]
Comment (Fosol) [reply app/subscriber/src/features/my-products/MyProducts.tsx:20]: hopefully not, but if they did it would make a request each time :D

**Review by Fosol — APPROVED** [review]

**Pull request merged**